geoutils/raster/raster.py

```diff
@@ -1523,6 +1529,7 @@ def set_nodata(
 
         # Update the nodata value
         self._nodata = new_nodata
+        self.data.fill_value = new_nodata
```

Comment:
This little addition was necessary to make `raster_equal` happy when `set_nodata` is called!